WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsRepository.kt

```diff
@@ -224,37 +234,45 @@ class AnalyticsRepository @Inject constructor(
         else -> DeltaPercentage.Value((round((currentVal - previousVal) / previousVal) * ONE_H_PERCENT).toInt())
     }
 
-    private suspend fun fetchStats(
+    private fun shouldUpdatePreviousStats(startDate: String, endDate: String) =
+        (previousRevenueStats == null || previousRevenueStats?.result?.isCompleted == true) &&
```

Comment:
I see you are updating the result every time it's completed, can you explain why?

From my perspective, unless there was a force refresh, even if the result is `completed`, it's still a valid one, and there is no need to go down the layers, WDYT?

Reply:
I was thinking about that for a while, let me explain my thoughts:

Force refresh is received in the repository as true when a user does a pull to refresh action. So, each time user does it I would receive 3 requests (One per card, so, Revenue, Products, and Orders) to force update data, If I would be using the force=true to decide when should I update the result, each card request would do it so I would be doing 3 same requests to the network which is exactly what we were trying to avoid. With this flow, the force refresh is being sent to fluxC only for the first call and the other 2 are waiting until it finishes and the result is received, so, it's avoiding that 2 last calls. Next time a force request is done, only the first call will be over the network, updating the results and so on..

Does it make sense for you?

Reply:
I didn't mean check only the `force` value, but combine with `isCompleted` to decide.
But actually I didn't check your code very well, it has an issue, since you have an `&&` operator there, unless we change the `date`, when we try to refresh, we will keep using the cached value.

I think a bit of code would make what I meant clearer:

```kotlin
private fun shouldUpdateCurrentStats(force: Boolean, startDate: String, endDate: String) =
        currentRevenueStats?.startDate != startDate || currentRevenueStats?.endDate != endDate
             || (force && currentRevenueStats?.result?.isCompleted == true)
```

If `currentRevenueStats` is null, then the first check will be `true`, but if itsn't, then we will check the dates first, and if they are the same, we won't refresh the result unless we need to force refresh, and with the `isCompleted` check, we can be sure we won't invalidate it while it's still running.

Reply:
I still don't see the point about the force there I will test the code :). Thanks for the suggestion!
 
```suggestion
        (previousRevenueStats == null || previousRevenueStats?.result?.isCompleted == true) &&
```

Reply:
I understand your point now with the debugger and stopping in the calls. 
It seems clearest and safest.